File: src/Caja_Chica/WEBSERVICE/clienteStructura.py
import barrister

trans  = barrister.HttpTransport("http://localhost:7667/add")

# automatically connects to endpoint and loads IDL JSON contract
client = barrister.Client(trans)

print client.Calculator.add({'clase':'ok','metodo':'restar','codusuarios':['1','2']})


print
print "IDL metadata:"
meta = client.get_meta()
for key in [ "barrister_version", "checksum" ]:
    print "%s=%s" % (key, meta[key])

# not printing this one because it changes per run, which breaks our
# very literal 'examples' test harness, but let's verify it exists at least..
assert meta.has_key("date_generated")

File: src/Caja_Chica/WEBSERVICE/clienteStructura.php
<?php
include_once("barrister.php");

$barrister = new Barrister();
$client    = $barrister->httpClient("http://localhost:7667/william");
$Calculator      = $client->proxy("Calculator");
$array = array(
    "clase" => "ok",
    "metodo" => "restar",
    "codusuarios" => array("foo", "bar", "hallo", "world")
);

print_r ($Calculator->add($array));

echo "\nIDL metadata:\n";
$meta = $client->getMeta();
$keys = array("barrister_version", "checksum");
foreach ($keys as $i=>$key) {
  echo "$key=$meta[$key]\n";
}

?>
